R/QCplots.R

```diff
@@ -71,160 +43,288 @@
 
-        if (!tolower(plot_type) == "histogram") {
-            p <- ggplot(qcdata, aes_string(x = "Sample", y = metric, group = 1))
-        }
-
-        p <- switch(tolower(plot_type),
-                    bar = {p + geom_bar(stat = "identity", color = barColor, fill = barFill, alpha = barAlpha, width = barWidth)},
-                    point = {p + geom_point(color = pointColor, fill = pointFill, shape = pointShape, alpha = pointAlpha, size = pointSize)},
-                    pointline = {p +
-                            geom_point(color = pointColor, fill = pointFill, shape = pointShape, alpha = pointAlpha, size = pointSize) +
-                            geom_line(color = lineColor, size = lineSize, linetype = lineType, alpha = lineAlpha)},
-                    histogram = {p <- ggplot(qcdata, aes_string(x = metric)) +
-                        geom_histogram(colour = histColor, fill = histFill, size = histSize, alpha = histAlpha)}
-        )
-
-        # Draw hline xSD above or below the mean
+        metricMedian <- median(as.numeric(thisMetric[-1]), na.rm = TRUE)
+        metricMean   <- mean(as.numeric(thisMetric[-1]), na.rm = TRUE)
+        metricSD     <- sd(as.numeric(thisMetric[-1]), na.rm = TRUE)
         SD <- metricSD * hlineSD
-        if (hlineSD > 0 & tolower(plot_type) == "histogram") {
 
-            # Plot vlines for the histogram
-            p <- p +
-                geom_vline(xintercept = metricMedian, color = "grey70") +
-                geom_vline(xintercept = metricMean + SD, color = "firebrick3", linetype = "longdash") +
-                geom_vline(xintercept = metricMean - SD, color = "firebrick3", linetype = "longdash")
-        } else if (hlineSD > 0) {  # Use hlines for other plot types
-            p <- p +
-                geom_hline(yintercept = metricMedian, color = "grey70") +
-                geom_hline(yintercept = metricMean + SD, color = "firebrick3", linetype = "longdash")
-            # Most qc plots floor to 0 so no point plotting -SD if it goes below zero.
-            if (metricMean - SD > 0) {
-                p <- p + geom_hline(yintercept = metricMean - SD, color = "firebrick3", linetype = "longdash")
+        #aesthetics
+        color = "dodgerblue3"
+
+        if (plotType == "canvasxpress") {
+            cx.data <- qcdata %>%
+                dplyr::select(Sample, !!rlang::sym(metric)) %>%
+                dplyr::arrange(Sample) %>%
+                tibble::column_to_rownames("Sample") %>%
+                t() %>%
+                as.data.frame()
+
+            decorations = list()
+
+            if (hlineSD > 0 & plot_metric == "histogram") {
+                decorations <- .getCxPlotDecorations(decorations = decorations,
+                                                     color = "grey70",
+                                                     width = 2,
+                                                     x = metricMedian
+                                                     )
```

Comment:
there is inconsistency style for closing `)` with blow lines.
In addition,  for me,  I will add add at the end of the line not in a separate line in the `else` section